File: pygSellerGoodsInterface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.TbBrand;

/**
 * 品牌service
 * **/
public interface BrandService {
	
	/**
	 * 查询所有点品牌
	 * **/
	public List<TbBrand> getAll () ;
	
	/**分页查询品牌
	 * @param pageNub : 当前页码
	 * @param pageSize : 每页数量
	 * **/
	public PageResult findPage(int pageNub ,int pageSize);
	
	
	/**分页查询品牌带条件
	 * @param pageNub : 当前页码
	 * @param pageSize : 每页数量
	 * **/
	public PageResult findPage(TbBrand brand,int pageNub ,int pageSize);
	
	
	/**
	 * 新增品牌
	 * **/
	public void add (TbBrand brand);
	
	/**
	 * 查询一个品牌
	 * **/
	public TbBrand findOne (Long id);
	
	/**
	 * 编辑品牌
	 * **/
	public void update (TbBrand brand);
	
	/**删除品牌**/
	public void delete (Long [] ids);

	public  List<Map<String, Object>>  selectOptionList();
}

File: pygManagerWeb/src/main/webapp/js/service/brandService.js

app.service("brandService",function ($http) {
		this.findall = function () {
			return $http.get('../brand/getAll.do');
		}
		
		this.search = function (page,size,search) {
			return $http.post('../brand/search.do?page='+page+'&size='+size,search);
		}
		
		this.findOne = function (id) {
			return $http.get('../brand/findOne.do?id=' +id);
		}
		
		this.deletebrand = function (ids) {
			return $http.get('../brand/delete.do?ids=' +ids);
		}
		
		this.add = function (entity) {
			return $http.post('../brand/add.do' , entity);
		}
		this.update = function (entity) {
			return $http.post('../brand/update.do' , entity);
		}
			
});